plugins/catalog-backend/src/permissions/rules/index.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { hasAnnotation } from './hasAnnotation';
+import { isEntityKind } from './isEntityKind';
+import { isEntityOwner } from './isEntityOwner';
+
+/**
+ * These permission rules can be used to conditionally filter catalog entities
+ * or describe a user's access to the entities.
+ * @public
+ */
+export const PermissionRules = {
+  hasAnnotation,
+  isEntityKind,
+  isEntityOwner,
```

Comment:
I think this is probably a good moment to have a brief think about whether it's worth implementing a few more rules for early adopters to use. A couple come to mind for me:

- `hasLabel`, with similar implementation to `hasAnnotation` but based on [`metadata.labels`](https://backstage.io/docs/features/software-catalog/descriptor-format#labels-optional). This type of rule is useful for fields inside metadata and spec that contain arrays of values.
- Generic rules `hasMetadata` and `hasSpec` (or even `hasProperty` covering both) could be worth exploring as an alternative to implementing rules for every single-value field in metadata and spec.

Reply:
> rules that we know we will definitely need for the scope of this integration

I think the tricky bit here is that we really don't need _any_ rules for the _integration_ to work correctly. The rules are only there as a toolbox for policy authors to return conditional responses, so to an extent we need to be speculative about which rules we include. There wasn't a ton of thought put into which rules we included in the PRFC branch, so I suggest we don't put too much store in what is and isn't present in that branch.

No real preference about PR organization, but we shouldn't consider the catalog integration to be done until we've thought about whether we have enough initial rules in place. As such it might be easier to just get all the rules we want in under this PR, rather than having to circle back.

Reply:
I thought it would be good to break out the rules into a separate PR, so I've put up #8766 and have rebased this PR on top of that one. Let's take this discussion there!